prefect_dbt/cli/commands.py

```diff
@@ -157,3 +164,193 @@ def trigger_dbt_cli_command_flow():
+        ),
+    )
+    overwrite_profiles: bool = Field(
+        default=False,
+        description=(
+            "Whether the existing profiles.yml file under profiles_dir "
+            "should be overwritten with a new profile."
+        ),
+    )
+    dbt_cli_profile: Optional[DbtCliProfile] = Field(
+        default=None,
+        description=(
+            "Profiles class containing the profile written to profiles.yml. "
+            "Note! This is optional and will raise an error if profiles.yml already "
+            "exists under profile_dir and overwrite_profiles is set to False."
+        ),
+    )
+
+    @root_validator(pre=True)
+    def _check_installation(cls, values):
+        """
+        Check that dbt is installed.
+        """
+        if not which("dbt"):
+            raise ImportError(
+                "dbt-core needs to be installed to use this task; run "
+                '`pip install "prefect-dbt[cli]"'
+            )
+        return values
+
+    @root_validator
+    def _check_profiles(cls, values):
+        """
+        Check that the profiles_dir and dbt_cli_profile are valid.
+        """
+        profiles_dir = values.get("profiles_dir")
+        if profiles_dir is None:
+            if values.get("env", {}).get("DBT_PROFILES_DIR") is not None:
+                profiles_dir = values["env"]["DBT_PROFILES_DIR"]
+            else:
+                profiles_dir = os.getenv("DBT_PROFILES_DIR", Path.home() / ".dbt")
+        profiles_dir = Path(profiles_dir).expanduser()
+        values["profiles_dir"] = profiles_dir
+        print(profiles_dir)
+
+        # https://docs.getdbt.com/dbt-cli/configure-your-profile
+        # Note that the file always needs to be called profiles.yml,
+        # regardless of which directory it is in.
+        profiles_path = profiles_dir / "profiles.yml"
+        overwrite_profiles = values.get("overwrite_profiles")
+        dbt_cli_profile = values.get("dbt_cli_profile")
+        if not profiles_path.exists() or overwrite_profiles:
+            if dbt_cli_profile is None:
+                raise ValueError(
+                    "Since overwrite_profiles is True or profiles_path is empty, "
+                    "need `dbt_cli_profile` to write a profile"
+                )
+            profile = dbt_cli_profile.get_profile()
+            profiles_dir.mkdir(exist_ok=True)
+            with open(profiles_path, "w+") as f:
+                yaml.dump(profile, f, default_flow_style=False)
+        elif dbt_cli_profile is not None:
+            raise ValueError(
+                f"Since overwrite_profiles is False and profiles_path {profiles_path} "
+                f"already exists, the profile within dbt_cli_profile couldn't be used; "
+                f"if the existing profile is satisfactory, do not set dbt_cli_profile"
+            )
+        return values
+
+    @root_validator
+    def _process_commands(cls, values):
```

Comment:
Verifying that at least one command is dbt related make sense, but the modification of the commands I think should happen directly prior to invoking the commands for similar reasons mention in the comment on the validator above this one.